app/models/history/effect_factory.rb

```diff
@@ -20,11 +20,15 @@ def create!(type_name, account_public_key, details)
       where(address: Stellar::Convert.pk_to_address(account_public_key)).
       first
 
+    type = History::Effect::BY_NAME[type_name]
```

Comment:
Yeah, that's much better.  Thanks!  I always forget `fetch` raises an error... I tend to just use it for the default arg.